test/tftp/tftp_server_test.rb

```diff
@@ -0,0 +1,104 @@
+require 'test_helper'
+require 'tftp/server'
+require 'tftp/tftp_plugin'
+
+module TftpGenericServerSuite
+  def setup
+    @rootdir = "/some/root"
+    @mac = "aa:bb:cc:dd:ee:ff"
+    @content = "file content"
+    @file_handle = mock
+    @file_handle.stubs(:write).with(@content)
+    @file_handle.stubs(:readlines).returns(@content)
+    Proxy::TFTP::Plugin.settings.stubs(:tftproot).returns(@rootdir)
+    setup_paths
+  end
+
+  def test_set
+    FileUtils.stubs(:mkdir_p).returns(true)
```

Comment:
If you want to use mocks for these tests, can you extract file-related operations into methods in ```Proxy::TFTP::Server``` class? For example, around https://github.com/lzap/smart-proxy/blob/pxegrub2/modules/tftp/server.rb#L10 would become something like:

```
def write_config_for_mac(dir, file, config)
  FileUtils.mkdir_p dir
  File.open(File.join(dir, file), 'w') {|f| f.write(config) }
end
``` 
etc.

Such a refactor would both improve readability (both code and tests) and reduce the fragility of mocks...